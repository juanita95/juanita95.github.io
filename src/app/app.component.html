<router-outlet></router-outlet>
<ngx-loading [show]="loadingService.loading"
    [config]="{animationType: loadingService.ngxLoadingAnimationTypes.circleSwish, primaryColour: this.loadingService.loadingColor, backdropBorderRadius: '13px'}">
</ngx-loading>
