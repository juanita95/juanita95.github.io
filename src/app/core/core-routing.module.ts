import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CoreComponent } from './core.component';
import { LoginComponent } from './login/login.component';
import { TokenComponent } from './token/token.component';

const routes: Routes= [
  {
    path:'',
    component: CoreComponent,
    children:[ 
      {
        path:'login',
        component: LoginComponent
      },
      {
        path:'token',
        component: TokenComponent,
      }
    ]
  }
]

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class CoreRoutingModule { }
