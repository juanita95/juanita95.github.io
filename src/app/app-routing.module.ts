import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { AuthGuard } from './shared/guards/auth.guard';

const routes: Routes = [
  {
    path: 'auth',
    loadChildren: () => import('./core/core.module').then(m => m.CoreModule)
  },
  {
    path: 'spoty-app',
    canActivate: [AuthGuard],
    loadChildren: () => import('./pages/pages.module').then(m => m.PagesModule)
  },
  {
    path: '**',
    redirectTo: '/auth/login',
  }

];


@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
