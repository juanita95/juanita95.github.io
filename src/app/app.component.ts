import { Component } from '@angular/core';
import { LoadingService } from './shared/services/loading.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'myMusicApp';
  jsonLdObject = {
    "@context": "https://schema.org/",
    "@type": "WebSite",
    "name": "juanitafy",
    "url": "https://juanita95.github.io/juanitafy/",
    "potentialAction": {
      "@type": "SearchAction",
      "target": "https://juanita95.github.io/juanitafy/{search_term_string}atenea",
      "query-input": "required name=search_term_string"
    }
  };

  constructor(
    public loadingService: LoadingService) {
  }

  get jsonLdString(): string {
    return JSON.stringify(this.jsonLdObject);
  }
}
