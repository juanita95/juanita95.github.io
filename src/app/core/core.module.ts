import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { CoreRoutingModule } from './core-routing.module';
import { LoginComponent } from './login/login.component';
import { ComponentsModule } from '../components/components.module';
import { TokenComponent } from './token/token.component';
import { CoreComponent } from './core.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    LoginComponent,
    TokenComponent,
    CoreComponent
  ],
  imports: [
    RouterModule,
    ComponentsModule,
    CommonModule,
    CoreRoutingModule
  ]
})
export class CoreModule { }
