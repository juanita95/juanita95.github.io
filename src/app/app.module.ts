import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MaterialDesignerModule } from './material-designer/material-designer.module';
import { ComponentsModule } from './components/components.module';
import { PagesModule } from './pages/pages.module';
import { CoreModule } from './core/core.module';
import { ActionReducer, MetaReducer, StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { Interceptor } from './shared/services/interceptor.service';
import { ROOT_REDUCERS } from './ngxr/app.state';
import { EffectsModule } from '@ngrx/effects';
import { TrackEffect } from './ngxr/track/track.effects';
import { AuthEffect } from './ngxr/auth/auth.effects';
import { localStorageSync } from 'ngrx-store-localstorage';

export function localStorageSyncReducer(
  reducer: ActionReducer<any>
): ActionReducer<any> {
  return localStorageSync({
    keys: ['tracks', { auth: ['accessToken']}],
    rehydrate: true,
  })(reducer);
}
const metaReducers: Array<MetaReducer<any, any>> = [localStorageSyncReducer];

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    BrowserAnimationsModule,
    MaterialDesignerModule,
    ComponentsModule,
    CoreModule,
    PagesModule,
    StoreModule.forRoot(ROOT_REDUCERS, {metaReducers}),
    StoreDevtoolsModule.instrument({ name: 'TEST' }),
    EffectsModule.forRoot([TrackEffect, AuthEffect])
  ],
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: Interceptor,
      multi: true,
    },
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
