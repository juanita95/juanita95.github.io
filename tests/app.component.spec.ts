import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const appComponent = fixture.componentInstance;
    expect(appComponent).toBeTruthy();
  });

  it(`should have as title 'myMusicApp'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const appComponent = fixture.componentInstance;
    expect(appComponent.title).toEqual('myMusicApp');
  });

  it(`should make match with snapshot'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled).toMatchSnapshot()
  });

});
