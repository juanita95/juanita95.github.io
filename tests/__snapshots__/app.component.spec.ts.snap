// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should make match with snapshot' 1`] = `
<div
  id="root2"
>
  <router-outlet />
  
  <ngx-loading />
</div>
`;
